devcontainers/build.py

```diff
@@ -0,0 +1,223 @@
+                self._decorate_cmake_variable('CMAKE_BUILD_TYPE', config)
+            ]
+            print(f'running configure command for CMake config {config}')
+            self._run(command)
+
+    def build(self: 'Routines') -> None:
+        if not self._params.build_directory.is_dir():
+            sys.exit(f'build directory \'{self._params.build_directory}\' was not found')
+
+        print(f'using {self._params.cores} threads')
+        for config in self._params.build_configs:
+            directory = self._params.build_directory.joinpath(config)
+            print(f'building for CMake configuration \'{config}\'')
+            self._run([
+                'cmake',
+                '--build', str(directory), 
+                '--parallel', str(self._params.cores)
+            ])
+
+    def link(self: 'Routines') -> None:
+        if 'Debug' in self._params.build_configs:
+            path = self._params.build_directory.joinpath('Debug').joinpath('compile_commands.json')
+            print(f'creating symlink for path \'{path}\'')
+        if 'Release' in self._params.build_configs:
+            path = self._params.build_directory.joinpath('Release').joinpath('compile_commands.json')
+            print(f'creating symlink for path \'{path}\'')
+        if path is None:
+            sys.exit('no supported CMake configs detected')
+        symlink = pathlib.Path.cwd().joinpath('compile_commands.json')
+        if symlink.is_symlink():
+            symlink.unlink()
+        pathlib.Path.cwd().joinpath('compile_commands.json').symlink_to(path)
+
+    def _run(self: 'Routines', command: typing.List[str]) -> None:
+        print('running command: ' + ' '.join(command))
+        code = subprocess.run(command, encoding='UTF-8', stderr=subprocess.STDOUT, env=os.environ).returncode
+        if code:
+            sys.exit(f'error: subprocess failed: {errno.errorcode[code]} (code: {code})')
+
+    def _decorate_cmake_variable(self: 'Routines', var: str, value: str, type: typing.Union[str, None] = None) -> str:
+        if type is not None:
+            return f'-D{var.upper()}:{type}={value}'
+        return f'-D{var.upper()}={value}'
+    
+    def _decorate_conan_profile_entry(self: 'Routines', section: str, entry: str, value: str) -> str:
+        return f'--{section}={entry}={value}'
+
+
+def parse_cli_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=description, formatter_class=argparse.RawDescriptionHelpFormatter)
+    # Routines
+    parser.add_argument('-b', '--build', action='store_true', dest='build',
+        help='runs cmake build, runs after configure.'
+    )
+    parser.add_argument('-c', '--configure', action='store_true', dest='configure',
+        help='runs cmake configure. If CMakeUserPresets.json is preset, uses conan presets. '
+        'Runs after install.'
+    )
+    parser.add_argument('-i', '--install', action='store_true', dest='install',
```

Comment:
At first, I thought you would "install" Artery here, but actually it is "conan install". I think you should rename it for clarity.